cmake/modules/yaml.cmake

```
@@ -364,22 +426,67 @@ function(yaml_save)
   if(NOT yaml_file)
     zephyr_check_arguments_required(${CMAKE_CURRENT_FUNCTION} ARG_YAML FILE)
   endif()
-
-  zephyr_get_scoped(json_content ${ARG_YAML_NAME} JSON)
-  to_yaml("${json_content}" 0 yaml_out)
-
   if(DEFINED ARG_YAML_FILE)
     set(yaml_file ${ARG_YAML_FILE})
   else()
     zephyr_get_scoped(yaml_file ${ARG_YAML_NAME} FILE)
   endif()
+
+  zephyr_get_scoped(genex ${ARG_YAML_NAME} GENEX)
+  zephyr_get_scoped(json_content ${ARG_YAML_NAME} JSON)
+  to_yaml("${json_content}" 0 yaml_out ${genex})
+
   if(EXISTS ${yaml_file})
     FILE(RENAME ${yaml_file} ${yaml_file}.bak)
   endif()
   FILE(WRITE ${yaml_file} "${yaml_out}")
+
+  set(save_target ${ARG_YAML_NAME}_yaml_saved)
+  if (NOT TARGET ${save_target})
+    # Create a target for the completion of the YAML save operation.
+    # This will be a dummy unless genexes are used.
+    add_custom_target(${save_target} ALL DEPENDS ${yaml_file})
+    set_target_properties(${save_target} PROPERTIES
+      genex_save_count 0
+      temp_files ""
+    )
+  endif()
+
+  if (genex)
+    get_property(genex_save_count TARGET ${save_target} PROPERTY genex_save_count)
+    if (${genex_save_count} EQUAL 0)
+      # First yaml_save() for this context with genexes enabled
+      add_custom_command(
+        OUTPUT ${yaml_file}
+        DEPENDS $<TARGET_PROPERTY:${save_target},json_file>
+        COMMAND ${CMAKE_COMMAND}
+                -DJSON_FILE="$<TARGET_PROPERTY:${save_target},json_file>"
+                -DYAML_FILE="${yaml_file}"
+                -DTEMP_FILES="$<TARGET_PROPERTY:${save_target},temp_files>"
+                -P ${ZEPHYR_BASE}/cmake/yaml-filter.cmake
+      )
+    endif()
+
+    math(EXPR genex_save_count "${genex_save_count} + 1")
+    set_property(TARGET ${save_target} PROPERTY genex_save_count ${genex_save_count})
+
+    cmake_path(SET yaml_path "${yaml_file}")
+    cmake_path(GET yaml_path STEM yaml_file_no_ext)
+    set(json_file ${yaml_file_no_ext}_${genex_save_count}.json)
+    set_property(TARGET ${save_target} PROPERTY json_file ${json_file})
+
+    # comment this to keep the temporary JSON files
+    get_property(temp_files TARGET ${save_target} PROPERTY temp_files)
+    list(APPEND temp_files ${json_file})
+    set_property(TARGET ${save_target} PROPERTY temp_files ${temp_files})
```

this can be simplified to:
```suggestion
    set_property(TARGET ${save_target} APPEND PROPERTY temp_files ${json_file})
```